xwiki-platform-core/xwiki-platform-security/xwiki-platform-security-authorization/xwiki-platform-security-authorization-api/src/main/java/org/xwiki/security/authorization/Right.java

```diff
@@ -344,7 +368,9 @@ private Set<Right> cloneImpliedRights(Set<Right> impliedRights)
      */
     public static List<Right> values()
     {
-        return UNMODIFIABLE_VALUES;
+        return VALUES.stream()
+            .filter(Objects::nonNull)
+            .collect(Collectors.toUnmodifiableList());
```

Comment:
Are you sure that callers don't rely on positions in the returned list?

Reply:
Hmmm I cannot guarantee it for usages of this API outside of XS. And for XS itself it's used internally by `Right#size` which is used by `RightSet` and `RightMap`. So probably a mistake.

Reply:
So I see various options: 
  1. We juste return the unmodifiable list of values. The problem is that external callers might not check for null and get a NPE because of that.
  2. We return a list of values without null, and we fix  `#size` to rely on `VALUES`. The problem is that it create a bit of inconsistency as `#size` won't return the same size as this.
  3. We return a list of values without null, we introduce a new API that returns the null values, and we use that one when needing to check the size in XS for RightSet/RightMap. 

Personally I think 3 is the safest.

Reply:
Maybe we should decouple the concept of "size" from the concept of "maximum value of a right" to make things clearer for callers.

Reply:
> Maybe we should decouple the concept of "size" from the concept of "maximum value of a right" to make things clearer for callers.

Well I thought about adding a new API just for the size but it feels more limiting than entirely returning the actual list of values with null elements.